headers.go

```diff
@@ -115,15 +115,55 @@ func (h ProtectedHeader) SetType(typ any) (any, error) {
 
 // SetCWTClaims sets the CWT Claims value of the protected header.
 func (h ProtectedHeader) SetCWTClaims(claims CWTClaims) (CWTClaims, error) {
-	iss, hasIss := claims[1]
-	if hasIss && !canTstr(iss) {
-		return claims, errors.New("cwt claim: iss: require tstr")
-	}
-	sub, hasSub := claims[2]
-	if hasSub && !canTstr(sub) {
-		return claims, errors.New("cwt claim: sub: require tstr")
+	for name, _ := range claims {
+		switch name {
+		case 1:
+			iss, hasIss := claims[name]
+			if hasIss && !canTstr(iss) {
+				return claims, errors.New("cwt claim: iss: require tstr")
+			}
+		case 2:
+			sub, hasSub := claims[name]
+			if hasSub && !canTstr(sub) {
+				return claims, errors.New("cwt claim: sub: require tstr")
+			}
```

Comment:
Since you are using a `for` loop against `claims CWTClaims` where `type CWTClaims map[any]any`, you don't need to do something like `iss, hasIss := claims[name]` as `hasIss` is always true.

You may consider a simpler and better code style. Also, you should use constants instead of `1`, `2`, ..., for readability and maintainability.
```suggestion
	for name, value := range claims {
		switch name {
		case CWTClaimIssuer:
			if !canTstr(value) {
				return claims, errors.New("cwt claim: iss: require tstr")
			}
		case CWTClaimSubject:
			if !canTstr(value) {
				return claims, errors.New("cwt claim: sub: require tstr")
			}
```
So as for the rest of the code.